### PR TITLE
Enable CI for pull requests; fix project setup

### PR DESCRIPTION
**Pull request by bluenote10**
The goal of this PR is to enable the CI checks for pull requests to allow adding further checks (type checker, black) in follow-up PRs as a preparation for #77.

While doing so, I've also added a weekly re-run of the CI. This can be useful to see if there are regressions introduced from changes in upstream dependencies (because the CI strategy of this repo is not to fully pin the dependencies, but rather test against a "moving target").

Enabling the checks however revealed that the CI is currently broken, due to a violation in the `pyproject.toml` vs `setup.py` setup. Basically executing `pip install ".[all]"` fails with modern versions of pip with errors like these:

```
********************************************************************************
The following seems to be defined outside of `pyproject.toml`:

`classifiers = ['License :: OSI Approved :: Apache Software License', 'Topic :: Multimedia :: Sound/Audio', 'Topic :: Scientific/Engineering']`

According to the spec (see the link below), however, setuptools CANNOT
consider this value unless `classifiers` is listed as `dynamic`.

https://packaging.python.org/en/latest/specifications/pyproject-toml/#declaring-project-metadata-the-project-table

To prevent this problem, you can list `classifiers` under `dynamic` or alternatively
remove the `[project]` table from your file and rely entirely on other means of
configuration.
********************************************************************************
```

Basically pip is complaining about the redundancy that currently exists between the information in the `pyproject.toml` vs `setup.py`.

I have resolved this issues by removing the information from the `pyproject.toml` -- note that it is fully redundant to the information specified in `setup.py`.

An alternative would be to resolve the redundancy towards `pyproject.toml` instead, i.e., basically remove the `setup.py` altogether. This should be feasible as well. If you prefer, I could also try to resolve the redundancy in this direction.

### CODE REVIEW

**Comment by davidmarttila** [comment]
@csteinmetz1 Would it be possible to merge this pull request? Or to consider alternatives like declaring `license`, `requires-python` etc as `dynamic` in `pyproject.toml`, or as the pull request suggests, dropping `setup.py` altogether.

Currently, the main branch of the repository is broken and can't be installed from the repository, i.e.

```
pip install "auraloss @ git+https://github.com/csteinmetz1/auraloss"
```

or 

```
git clone https://github.com/csteinmetz1/auraloss
cd auraloss && pip install .
```

fail with the error message mentioned in the PR.

**Comment by csteinmetz1** [comment]
Hey, I'd like to merge this PR. @bluenote10, is this still the right solution going forward, or is there anything else we need to consider now?

**Comment by bluenote10** [comment]
> is this still the right solution going forward, or is there anything else we need to consider now?

It worked back then when I'd tested it. Enabling the CI pipeline for PRs should give a definite answer :wink:
